pkg/build/controller/build/build_controller.go

```diff
@@ -1438,35 +1460,23 @@ func (bc *BuildController) createBuildCAConfigMapSpec(build *buildv1.Build, buil
 	cm := &corev1.ConfigMap{
 		ObjectMeta: metav1.ObjectMeta{
 			Name: buildapihelpers.GetBuildCAConfigMapName(build),
-			Annotations: map[string]string{
-				"service.alpha.openshift.io/inject-cabundle": "true",
-			},
 			OwnerReferences: []metav1.OwnerReference{
 				makeBuildCAOwnerRef(buildPod),
 			},
 		},
+		Data: bc.additionalTrustedCAData,
 	}
-	if len(bc.additionalTrustedCAData) > 0 {
-		cm.Data = map[string]string{
-			buildutil.AdditionalTrustedCAKey: string(bc.additionalTrustedCAData),
+	if bc.injectServiceCA {
+		cm.Annotations = map[string]string{
+			"service.alpha.openshift.io/inject-cabundle": "true",
+		}
+		if cm.Data == nil {
+			cm.Data = make(map[string]string)
```

Comment:
For now, adding the `inject-cabundle` annotation is configurable so we can run the extended tests.